### ISSUE TITLE
Bug in send_message/3

### ISSUE THREAD

**Issue by blue-dinosaur** [BlakeWilliams/Elixir-Slack#160]
Hi, I suppose there is a bug on send message.

https://github.com/BlakeWilliams/Elixir-Slack/blob/58ffa73559b5cb29d41c2a472ee3c4ba53ed6333/lib/slack/sends.ex#L33

https://github.com/BlakeWilliams/Elixir-Slack/blob/58ffa73559b5cb29d41c2a472ee3c4ba53ed6333/lib/slack/sends.ex#L101

So send_message/3 calls open_im_channel/4 and passes an error handler with arity-1, but open_im_channel/4 calls it with no parameters, which leads to errors like this:

```
** (BadArityError) #Function<1.76340654/1 in Slack.Sends.send_message/3> with arity 1 called with no arguments
    (slack) lib/slack/sends.ex:101: Slack.Sends.open_im_channel/4
```

Is my understanding correct? is this a bug?

Thanks!

**Comment by BlakeWilliams** [comment]
Oh nice find, I think that may be a bug.

If you have interest, I'd love a PR with a test covering the issue.

**Comment by blue-dinosaur** [comment]
I was sending a message with the `display_name` of the user, which is what shows up on slack's sidebar.

If you look here:

https://github.com/BlakeWilliams/Elixir-Slack/blob/58ffa73559b5cb29d41c2a472ee3c4ba53ed6333/lib/slack/lookups.ex#L10

You'll see that the lookup tests for `name`, which my case was different from `display_name`.

That means a `nil` value was showing up as `direct_message_id` here:

https://github.com/BlakeWilliams/Elixir-Slack/blob/58ffa73559b5cb29d41c2a472ee3c4ba53ed6333/lib/slack/sends.ex#L23-L36

When Lookups is called again on line 31 it does the same as in line 24 and returns `nil` again, so the slack im_open call will return an error.

We then get to the second bug, which is that on_error() is called with no arguments.

So. As for the first bug (whether display_name) should be check, it depends on whether this is intended or not. As for the second bug, I am not sure if passing the error from slack's api response is adequate -- should the Elixir-Slack user know about slack's internals?

I am submitting a patch.

**Comment by BlakeWilliams** [comment]
fixed in #161
